src/rest_vol_dataset.c

```diff
@@ -4363,3 +4393,111 @@ RV_convert_start_to_offset(hid_t space_id)
 
     return ret_value;
 } /* end RV_convert_start_to_offset() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_json_values_to_binary_callback
+ *
+ * Purpose:     A callback for RV_parse_response which will search
+ *              an HTTP response for the "value" field, and extract
+ *              the values into a newly allocated binary buffer.
+ *
+ *              Expects data in to be a pointer to an hid_t for the
+ *              datatype in the response, and data out to be the
+ *              address of a pointer that will point to the
+ *              newly allocated buffer.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ */
+static herr_t
+RV_json_values_to_binary_callback(char *HTTP_response, void *callback_data_in, void *callback_data_out)
+{
+
+    void      **out_buf    = (void **)callback_data_out;
+    hid_t       dtype_id   = *(hid_t *)callback_data_in;
+    yajl_val    parse_tree = NULL, key_obj;
+    char       *parsed_string;
+    herr_t      ret_value    = SUCCEED;
+    void       *value_buffer = NULL;
+    size_t      dtype_size   = 0;
+    H5T_class_t dtype_class  = H5T_NO_CLASS;
+
+#ifdef RV_CONNECTOR_DEBUG
+    printf("-> Converting response JSON values to binary buffer\n\n");
+#endif
+
+    if (!HTTP_response)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "HTTP response buffer was NULL");
+    if (!out_buf)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "output buffer was NULL");
+
+    if ((dtype_size = H5Tget_size(dtype_id)) <= 0)
```

Comment:
Comparing this < 0 will probably generate a warning on some compilers since it's unsigned